src/Centreon/Infrastructure/Downtime/DowntimeRepositoryRDB.php

```diff
@@ -0,0 +1,593 @@
+                AND srv.host_id = hosts.host_id
+            INNER JOIN `:db`.`contact`
+                ON contact.contact_alias = dwt.author'
+            . $aclRequest;
+
+        return $this->processRequest($request);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function findDowntimesByServiceForNonAdminUser(int $hostId, int $serviceId): array
+    {
+        if ($this->hasNotEnoughRightsToContinue()) {
+            return [];
+        }
+
+        // Internal call for non admin user
+        return $this->findDowntimesByService($hostId, $serviceId, false);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function findDowntimesByServiceForAdminUser(int $hostId, int $serviceId): array
+    {
+        // Internal call for an admin user
+        return $this->findDowntimesByService($hostId, $serviceId, true);
+    }
+
+    /**
+     *  Find all downtimes for a service (linked to a host).
+     *
+     * @param int $hostId Host id linked to this service
+     * @param int $serviceId Service id for which we want to find downtimes
+     * @param bool $isAdmin Indicates whether user is an admin
+     * @return array
+     * @throws \Exception
+     */
+    private function findDowntimesByService(int $hostId, int $serviceId, bool $isAdmin): array
+    {
+        $concordanceArray = [
+            // Relation for downtime
+            'id' => 'dwt.downtime_id',
+            'entry_time' => 'dwt.entry_time',
+            'is_cancelled' => 'dwt.cancelled',
+            'comment' => 'dwt.comment_data',
+            'deletion_time' => 'dwt.deletion_time',
+            'duration' => 'dwt.duration',
+            'end_time' => 'dwt.end_time',
+            'is_fixed' => 'dwt.fixed',
+            'start_time' => 'dwt.start_time',
+        ];
+
+        $this->sqlRequestTranslator->setConcordanceArray($concordanceArray);
+
+        $aclRequest = '';
+
+        if ($isAdmin === false) {
+            $aclRequest =
+                ' INNER JOIN `:dbstg`.`centreon_acl` acl
+                  ON acl.host_id = dwt.host_id
+                  AND acl.service_id = dwt.service_id
+                INNER JOIN `:db`.`acl_groups` acg
+                  ON acg.acl_group_id = acl.group_id
+                  AND acg.acl_group_activate = \'1\'
+                  AND acg.acl_group_id IN ('
+                . $this->accessGroupIdToString($this->accessGroups) . ')';
+        }
+
+        $request =
+            'SELECT SQL_CALC_FOUND_ROWS DISTINCT dwt.*, contact.contact_id AS author_id
+            FROM `:dbstg`.downtimes dwt
+            INNER JOIN `:dbstg`.hosts
+                ON hosts.host_id = dwt.host_id
+            INNER JOIN `:dbstg`.services srv
+                ON srv.service_id = dwt.service_id
+                AND srv.host_id = hosts.host_id
+                AND dwt.host_id = :host_id
+                AND srv.service_id = :service_id
+            INNER JOIN `:db`.`contact`
+                ON contact.contact_alias = dwt.author'
```

Comment:
If contact is deleted from centreon, we should be able to get the downtime information